### DISCUSSION TITLE
Bookmark/catalog cannot be seen in some pdf & chm files

### DISCUSSION THREAD

**Discussion by V-E-R-I-T-A-S** [sumatrapdfreader/sumatrapdf#4549] [Q&A] [answered]
In other pdf readers it's ok
[▲剪切流下细胞特异性滚动粘附力学研究.pdf](https://github.com/user-attachments/files/17095179/default.pdf)

**Comment by GitHubRulesOK** [comment, marked as answer]
It is not ok in all other readers (it has garbage at the end) so Tracker will attempt to repair it and mupdf can help too
 but it is still not perfect  however like this it works in all viewers OK  
[default part repaired.pdf](https://github.com/user-attachments/files/17097170/default.part.repaired.pdf)

**Reply by V-E-R-I-T-A-S** [reply]
Yes, the given pdf works well. But on the original file, I tried with Acrobat/Google Chrome/Microsoft Edge where no such problems happen. So I wonder why?

**Reply by GitHubRulesOK** [reply]
Edge useses either google or acrobat code so there should be only 2 to compare Acrobat work closely with Foxit who produced the code for Google Chromiums so not a lot of difference in some areas, either. 

Adobe Acrobat readers allow for mangled files and actually can refuse to open one built to their own specifications !! Actually google invoicing produce PDFs with odd headers but in this case its a wrong trailer, which Acrobat ignores using its internal editor, unless to a degree where reader will rewrite the file for you using that Acrobat editor functions.

**Reply by V-E-R-I-T-A-S** [reply]
Thank you for your answer! Then, what can i do to read such catalogs in sumatraPDF?

**Reply by GitHubRulesOK** [reply]
Try the fixed copy above but That was fixed in an editor but not all faults could be repaired from that source.

For example the working language is shown as `/Lang (›·ôÒ")` which is AFAIK not a language on Earth. 

Font as a rule will follow the numeric convention `/F0` `/F1` ... `/F7 ## 0 R/F8 ## 0 R` etc. whilst this file has `/F6 26 0 R/F7 28 0 R/FSec_song 11360 0 R` thus the 8th font has been poorly added as a name much later (overstamped bottom left?) it has NOT been embeded in the file as some others are!
![image](https://github.com/user-attachments/assets/0d57bcde-9ca9-4e06-950c-4d795356d01f)


There are other abnormalities so the file should end `%%EOF` however this one has a payload `%%EOF
transferOut:GTimeLimit.jsdxtsguser:Group.jsdxtsgcache:FalseClientIp:202.195.160.187ServerIp:192.188.4.30DateTime:2024-03-05 22:07:47„   wftrace`

I have not looked at all such errors just these are the most obvious.

hence I called it part fixed 
![image](https://github.com/user-attachments/assets/4d61a0ae-1f4a-4d68-a8bd-57e960b291d9)

![image](https://github.com/user-attachments/assets/391b2ab1-b902-4104-82a5-98aa39586169)
